source/4.0/guides/testing/factories.html.md

```diff
@@ -0,0 +1,140 @@
+---
+chapter: Testing
+title: Factories
+---
+
+When using ROM, you will need to test your code, and testing your code implies generating entities and use Factories to
+create records on your database. To do all of that, you can use `rom-factory`, which provides more or less same 
+functionnalities as `FactoryBot`.
+
+### Install RomFactory
+
+In your gemfile:
+
+```ruby
+gem 'rom-factory'
+```
+
+### Setup ROMFactory
+
+Then on your spec root file (like `spec_helper`), you have to initialize your factory:
+
+```ruby
+# Build database configuration
+db_config = { 
+  adapter: 'mysql2',
+  database: 'db_name',
+  host: 'database.host.io',
+  username: 'username',
+  password: 'password',
+  port: 3306
+}
+
+# Init your ROM container
+rom = ROM.container(:sql, db_config)do |config|
+        # Register ROM related files, especially Relations
+        config.auto_registration('../app/') # if your sources are in app folder
+      end
+
+# Declare your factory
+YourFactory = ROM::Factory.configure do |config|
+  config.rom = rom
+end
+
+# Load your factories' files
+Dir[App.root + '/spec/factories/**/*.rb'].each { |f| require f } # if you plan to put your factory files `spec/factories`
+```
+
+### Define your factory
+
+Then you can define your factories:
+
+```ruby
+YourFactory.define(:user) do |f|
+  f.sequence(:name) { |n| "User#{n}" }
+  f.website 'http://personal.website'
+  f.some_integer 42
+  f.created_at Time.now.utc
```

Comment:
Let's make generating the time a block, so each created record has a new timestamp. This is more realistic in testing since records created later in the spec will have an increasing created_at